src/makeInfinityAdder.js

```diff
@@ -4,7 +4,23 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let total = 0;
+
+  const adder = (value) => {
+    if (value !== undefined) {
+      total += value;
+
+      return adder;
+    } else {
+      const result = total;
+
+      total = 0;
+
+      return result;
+    }
+  };
+
+  return adder;
 }
 
 module.exports = makeInfinityAdder;
```

Comment:
The exported function name `makeInfinityAdder` should be changed to `makeAdder` to align with the task requirements.